tsinfer/inference.py

```diff
@@ -44,6 +45,32 @@
 logger = logging.getLogger(__name__)
 
 
+sample_data_time_metadata_definition = {
+    "description": "Time of an individual from the SampleData file.",
+    "type": "number",
+    # Defaults aren't currently used, see
+    # https://github.com/tskit-dev/tskit/issues/1073
+    "default": 0,
+}
+
+inference_type_metadata_definition = {
+    "description": "The type of inference used at this site.",
```

Comment:
Should we give a reference to the types we allow here?

Reply:
I've added an enum with a description of what the values mean. We should think about how to display documentation like this for a tree sequence.

Reply:
https://github.com/tskit-dev/tskit/issues/1092